src/Microsoft.Extensions.HealthChecks/IHealthCheckService.cs

```diff
@@ -0,0 +1,31 @@
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Extensions.HealthChecks
+{
+    /// <summary>
+    /// A service which can be used to check the status of <see cref="IHealthCheck"/> instances registered in the application
+    /// </summary>
+    public interface IHealthCheckService
+    {
+        /// <summary>
+        /// A <see cref="IReadOnlyDictionary{TKey, T}"/> containing all the health checks registered in the application.
+        /// </summary>
+        /// <remarks>
+        /// The key maps to the <see cref="IHealthCheck.Name"/> property of the health check, and the value is the <see cref="IHealthCheck"/>
+        /// instance itself.
+        /// </remarks>
+        IReadOnlyDictionary<string, IHealthCheck> Checks { get; }
+
+        /// <summary>
+        /// Runs all the health checks in the application and returns the aggregated status.
+        /// </summary>
+        /// <param name="cancellationToken">A <see cref="CancellationToken"/> which can be used to cancel the health checks</param>
+        /// <returns>
+        /// A <see cref="Task{T}"/> which will complete when all the health checks have been run,
+        /// yielding a <see cref="CompositeHealthCheckResult"/> containing the results.
+        /// </returns>
```

Comment:
Would you ever want more functionality here? Like the ability to invoke a single check by name? Wondering if this should be interface or class

Reply:
Yeah, the previous version had a lot of functionality on the [`IHealthCheckService` interface](https://github.com/dotnet-architecture/HealthChecks/blob/dev/src/Microsoft.Extensions.HealthChecks/IHealthCheckService.cs), I'm intentionally simplifying for the MVP. We need to revisit it prior to release though since adding to the interface would be breaking. Exposing the checks makes it possible for callers to invoke single checks if they want to (and we could provide helper extension methods). We could do grouping/tagging support in a similar way.

Reply:
I feel like these kinds of facades are usually better served as abstract classes. See `IAntiforgery` for something that's been a pain to deal with

Reply:
I'm not opposed to that idea.

Reply:
I want to dive down this rat hole a little bit (ping @davidfowl):

If we go with an abstract class, that's OK with me. I think we'd go with `HealthCheckService`(abstract) and `DefaultHealthCheckService`(concrete). That means to pull it out of DI you use `HealthCheckService` instead of an interface (which is not common in our system). Also, it makes me wonder if we even need the two classes, since the logic is so very simple (obviously we'd still use `virtual` in all the places).

The alternative is to keep it an interface but limit ourselves to adding functionality via extension methods (not ideal).

I think I lean towards the abstract class. Just because we haven't done it before doesn't mean it's the wrong approach here.